File: python/insertion_sort/insertion_sort.py
def insertion_sort(arr):

    # For every index in array
    for i in range(1, len(arr)):

        # Set current values and position
        currentvalue = arr[i]
        position = i

        # Sorted Sublist
        while position > 0 and arr[position - 1] > currentvalue:

            arr[position] = arr[position - 1]
            position = position - 1

        arr[position] = currentvalue


input_arr = [8, 4, 23, 42, 16, 15]
insertion_sort(input_arr)
print(f"Random input of [8, 4, 23, 42, 16, 15]. Sorted output -> {input_arr}")

reverse_sorted = [20, 18, 12, 8, 5, -2]
insertion_sort(reverse_sorted)
print(
    f"Sorted in reverse from input [20,18,12,8,5,-2]. Sorted output -> {reverse_sorted}"
)

nearly_sorted = [5, 12, 7, 5, 5, 7]
insertion_sort(nearly_sorted)
print(f"Nearly sorted from input [5,12,7,5,5,7]. Sorted output -> {nearly_sorted}")

File: built/javascript/algorithms/insertionSort/insertion-sort.js
'use strict';
// function insertionSort(array) {
//   const arr = array.slice();
//   for (let i = 1; i < arr.length; i++) {
//     for (let j = i; j > 0; j--) {
//       if (arr[j] < arr[j - 1]) {
//         [arr[j], arr[j - 1]] = [arr[j - 1], arr[j]];
//       } else {
//         break;
//       }
//     }
//   }
//   return arr;
// }
function insertionSort(int) {
    var arr = int.slice();
    for (var i = 1; i < arr.length; i++) {
        var j = i - 1;
        var temp = arr[i];
        while (j >= 0 && temp < arr[j]) {
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = temp;
    }
    return arr;
}
console.log(insertionSort([8, 4, 23, 42, 16, 15]));
console.log(insertionSort([20, 18, 12, 8, 5, -2]));
console.log(insertionSort([2, 3, 5, 7, 13, 11]));
